src/Avalonia.Controls/Flyouts/FlyoutBase.cs

```diff
@@ -0,0 +1,466 @@
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Right;
+                    break;
+
+                case FlyoutPlacementMode.Right: //Right & centered
+                    _popup.PlacementRect = new Rect(trgtBnds.Width - 1, 0, 1, trgtBnds.Height);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Right;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Right;
+                    break;
+
+                case FlyoutPlacementMode.RightEdgeAlignedBottom:
+                    _popup.PlacementRect = new Rect(trgtBnds.Width - 1, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.TopRight;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Right;
+                    break;
+
+                case FlyoutPlacementMode.Bottom: //Below & centered
+                    _popup.PlacementRect = new Rect(0, trgtBnds.Height - 1, trgtBnds.Width, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Bottom;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.BottomEdgeAlignedLeft:
+                    _popup.PlacementRect = new Rect(0, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomRight;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.BottomEdgeAlignedRight:
+                    _popup.PlacementRect = new Rect(trgtBnds.Width - 1, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.LeftEdgeAlignedTop:
+                    _popup.PlacementRect = new Rect(0, 0, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Left;
+                    break;
+
+                case FlyoutPlacementMode.Left: //Left & centered
+                    _popup.PlacementRect = new Rect(0, 0, 1, trgtBnds.Height);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Left;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Left;
+                    break;
+
+                case FlyoutPlacementMode.LeftEdgeAlignedBottom:
+                    _popup.PlacementRect = new Rect(0, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.TopLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.BottomLeft;
+                    break;
+
+                case FlyoutPlacementMode.Full:
```

Comment:
It's better to not include not supported enum values in the API.

Reply:
As this enum is carry-over from UWP should an exception be made in this case? Keeping the enum value in now ensures binary compatiblity at least so it's a form of future-proofing the design. It's entirely possible 'Full' is needed in the future (actually it's really useful for dialogs).

Reply:
Binary compatibility with what? UWPs enum? If we really wish to have that, we could leave a hole in the enumeration.

Reply:
Yes, with the UWP enum and any apps that might be using the int value itself for whatever reason. Would that affect many real-world apps? Probably not. I've just been conditioned to avoid making changes to enums especially in the middle (Full = 4). If this was at the end of the enum it wouldn't matter or if there was never any chance that Full would be supported in Avalonia. It just seems like a necessary future-proofing change to me.

Reply:
I would've like to implement `Full` but the positioning API for `Popup` is limited in that sense. It's the same reason I didn't include the `FlyoutShowOptions` which has an ExclusionRect (a Rect the popup tries to avoid displaying in), and a Point property (to show the Flyout at a specific point). This would all probably be doable with Overlay popups, but when their windowed it would require getting creative. 
For now, I'll just leave a hole in the enum where Full is.